docs/source/topics/blueprints.rst

```diff
@@ -0,0 +1,198 @@
+Blueprints
+==========
+
+
+Chalice blueprints are used to organize your application into logical
+components.  Using a blueprint, you define your resources and decorators in
+modules outside of your ``app.py``.  You then register a blueprint in your main
+``app.py`` file.  Any decorator supported on an application object is also
+supported in a blueprint.
+
+
+.. note::
+
+  The Chalice blueprints are conceptually similar to `Blueprints
+  <http://flask.pocoo.org/docs/latest/blueprints/>`__ in Flask.  Flask
+  blueprints allow you to define a set of URL routes separately from the main
+  ``Flask`` object.  This concept is extended to all resources in Chalice.  A
+  Chalice blueprint can have Lambda functions, event handlers, built in
+  authorizers, etc. in addition to a collection of routes.
+
+
+Example
+-------
+
+In this example we'll create a blueprint with part of our routes defined in a
+separate file.  First, let's create an application::
+
+    $ chalice new-project blueprint-demo
+    $ cd blueprint-demo
+    $ mkdir chalicelib
+    $ touch chalicelib/__init__.py
+    $ touch chalicelib/blueprints.py
+
+Next, we'll oen the ``chalicelib/blueprints.py`` file:
+
+.. code-block:: python
+
+    from chalice import Blueprint
+
+
+    extra_routes = Blueprint(__name__)
+
+
+    @extra_routes.route('/foo')
+    def foo():
+        return {'foo': 'bar'}
+
+
+The ``__name__`` is used to denote the import path of the blueprint.  This must
```

Comment:
One reason is that it keeps the API the same as Flask, allowing people to port their code over a bit easier.

A second reason is that a single package can occasionally have multiple blueprints.

Reply:
I was mainly more focused on why it is  a required parameter as opposed to being an optional parameter that set the `import_name` to `__name__` when not provided. That should still allow people to port it over from Flask pretty easily and set their own `import_name` if need be.

Reply:
There wasn't a strong reason it was required other than to mimic the Flask API.  I thought it would be good to make people aware of this parameter rather than to default it for them, but I don't have a strong reason why it needs to be required.